Read-only access to portfolio

Rundez opened the pull request:
### Your checklist for this pull request

- [ ] Make sure you are requesting to **pull a topic/feature/bugfix branch** (right side). Don't request your main!
- [ ] Make sure you are making a pull request against the **dev** branch (left side). Also you should start *your branch* off *dev*.
- [ ] Check the commit's or even all commits' message 
- [ ] Check if your code additions will fail linting checks
- [ ] Remember: Add PR description to [CHANGELOG](https://github.com/Puzzlepart/prosjektportalen365/blob/dev/CHANGELOG.md) with the ID that matches this PR

### Description

Added the possibility to see all projects when the user is in the "Porteføljeinnsyn" group. This can be toggled on and off in the projectlist web part. If it is toggled on, the user(s) in "Porteføljeinnsyn" will notice additional buttons appear right of the search bar. These buttons will change the current view, toggling between seeing all projects, and projects that the user is a member of.

### How to test

1. Add a user to the "Porteføljeinnsyn" group.
2. Navigate to the project list property pane and turn on "Vis alle prosjekter"
3. Toggle the buttons that appear. No deleted projects should appear, and projects you are not a member of should not ble clickable.

Disclaimer: 
Deleted projects will be shown if "All projects" is selected. A solution is will be delivered in a separate PR. 

### Relevant issues (if applicable)

fixes #443

💔Thank you!

olemp commented:
@Rundez Have you installed this somewhere for testing?

Rundez commented:
@olemp I have only served it in my own dev environment.

tarjeieo commented:
@Rundez can we setup a separate instance of pp365 in our demo tenant and share for testing? Seems like something that should be thoroughly tested

Rundez commented:
@tarjeieo I am setting one up.

Rundez commented:
Created a PP instance at [this](https://puzzlepart.sharepoint.com/sites/readOnlyPortfolio) site.

Rundez commented:
Two projects are created. The user Hugo First have been added in the "Porteføljeinnsyn" group, and will then have the ability to see projects he's not member of.

olemp reviewed (changes requested):
Most looks good, but I don't understand the logic 100%.

What does the "Vis alle prosjekter" toggle do? And how is it related to "Flisevisning"?

![image](https://user-images.githubusercontent.com/7606007/131849835-38f839cb-dfb0-4f40-9253-5c7fb71959cb.png)



Is it the toggle in web part properties that decides whether to show the "Mine prosjekter/Alle prosjekter" links? I don't find that so intuitive. Do we need a webpart property to toggle this option?

Also, we need to change the placeholder based on what "view" is selected.

![image](https://user-images.githubusercontent.com/7606007/131849778-b282c04c-baa1-4f53-990c-acbfa15b9a22.png)

Rundez commented:
Ah yes - it is not related to "Flisevisning" actually. As you said, it decides wether the links are displayed. It might be unnecessary, or it could be moved to another submenu in project properties. 

I agree with the placeholders. As I removed the function that checks for deleted projects the problem occured. I will create a fix.

Rundez commented:
Worked on wrong branch - that's why I merged and fuzzed. Could might have been handled better for readability.

- Removed the ability to hide the selection from webpart properties. This makes the functionality visible to everyone in Porteføljeinnsyn. 
- Added a placeholder base64 logo to projects which does not exist
- No access message rendered on all projects, this is now fixed.

Rundez commented:
Changed placeholder image when a user does not have access to the a project in collaboration with @tarjeieo. Did also confirm that it works the same way with the "Porteføljeinnsyn" group as "Porteføljeoversikt". 
![image](https://user-images.githubusercontent.com/38385042/135076392-c5077c6c-817c-4451-a098-9c44751ebed2.png)

Rundez commented:
> > > What is happening exactly? Seems to work here.
> > 
> > 
> > If it's over 20 projects, the project logo fetch is batched. It seems like that creates the issue. Not completely sure why it's happening though.
> 
> [This](https://github.com/Puzzlepart/prosjektportalen365/pull/498/commits/6757858b6f1f1fd29c303d717abe866e04c0ae32) might solve the issue.

Works like charm.
